components/locid/src/zerovec.rs

```diff
@@ -19,24 +19,27 @@
 //! To perform lookup, store the stringified locale in a canonical BCP-47 form as a byte array,
 //! and then use [`Locale::strict_cmp()`] to perform an efficient, zero-allocation lookup.
 //!
+//! To produce more human-readable serialized output, you can use [`UnvalidatedStr`].
+//!
 //! ```
 //! use icu_locid::Locale;
 //! use std::str::FromStr;
 //! use zerovec::ZeroMap;
+//! use zerovec::ule::UnvalidatedStr;
 //!
 //! // ZeroMap from locales to integers
-//! let data: &[(&[u8], u32)] = &[
-//!     (b"de-DE-u-hc-h12", 5),
-//!     (b"en-US-u-ca-buddhist", 10),
-//!     (b"my-MM", 15),
-//!     (b"sr-Cyrl-ME", 20),
-//!     (b"zh-TW", 25),
+//! let data: &[(&UnvalidatedStr, u32)] = &[
+//!     (b"de-DE-u-hc-h12".into(), 5),
```

Comment:
I'd prefer to do `str -> UnvalidatedStr` rather than `[u8] -> UnvalidatedStr`, these can be string literals. Also below.